### PR TITLE
[TypeDeclaration] Using ClassMethodReturnTypeOverrideGuard on ReturnTypeFromStrictParamRector

### PR DESCRIPTION
**Pull request by samsonasik**
Override return from parent is allowed if parent has no return, so same with other `ReturnTypeFromStrict*Rector`'s, this should use `ClassMethodReturnTypeOverrideGuard` as well.

### CODE REVIEW

**Comment by samsonasik** [comment]
The parent not exists check seems need to be checked, same with `ParentClassMethodTypeOverrideGuard`

File: packages/VendorLocker/NodeVendorLocker/ClassMethodReturnTypeOverrideGuard.php [hunk 1]
```diff
@@ -104,7 +104,7 @@ private function isReturnTypeChangeAllowed(ClassMethod $classMethod, Scope $scop
 
         // nothing to check
         if (! $parentClassMethodReflection instanceof MethodReflection) {
-            return true;
+            return ! $this->parentClassMethodTypeOverrideGuard->hasParentClassMethod($classMethod);
```
Comment (samsonasik) [inline packages/VendorLocker/NodeVendorLocker/ClassMethodReturnTypeOverrideGuard.php:107]: @staabm this is to tweak parent not found on no parent method usage on return type override check per return and has return true

https://github.com/rectorphp/rector-src/blob/7841dc582440c815796e703cf3cf9b3f17f80285/packages/VendorLocker/ParentClassMethodTypeOverrideGuard.php#L36-L38

https://github.com/rectorphp/rector-src/blob/7841dc582440c815796e703cf3cf9b3f17f80285/packages/VendorLocker/ParentClassMethodTypeOverrideGuard.php#L47

File: rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictParamRector/Fixture/parent_overridden.php.inc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+?>
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictParamRector\Fixture;
+
+use Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictParamRector\Source\ParentOverrideBase;
+
+class ParentOverridden extends ParentOverrideBase {
+    public function doFoo(ParentOverrideBase $param): ParentOverrideBase {
```
Comment (samsonasik) [inline rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictParamRector/Fixture/parent_overridden.php.inc:22]: this is allowed, see https://3v4l.org/UiFd6

File: rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictParamRector/Fixture/skip_parent_overridden.php.inc [hunk 3]
```diff
@@ -1,15 +0,0 @@
...
-
-class SkipParentOverridden extends SkipParentOverriddenBase {
-    public function doFoo(SkipParentOverridden $param) {
-        return $param;
-    }
-}
-
-
-class SkipParentOverriddenBase {
-    public function doFoo(SkipParentOverridden $param) {
```
Comment (staabm) [inline rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictParamRector/Fixture/skip_parent_overridden.php.inc:13]: I feel this test-fixture did miss a `SkipParentOverridden` incompatible return type at this point, e.g. `int`

File: rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictParamRector/Fixture/parent_overridden.php.inc (hunk 2 above)
Comment (staabm) [reply rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictParamRector/Fixture/parent_overridden.php.inc:22]: :+1:

File: rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictParamRector/Fixture/skip_parent_overridden.php.inc (hunk 3 above)
Comment (samsonasik) [reply rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictParamRector/Fixture/skip_parent_overridden.php.inc:13]: yes, it is about override return, the param is equal :)

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictParamRector/Fixture/skip_parent_overridden.php.inc (hunk 3 above)
Comment (staabm) [reply rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictParamRector/Fixture/skip_parent_overridden.php.inc:13]: what I try to say: we shouldn't delete this fixture, but keep it and just add a `:int` return type in the base method class

Comment (samsonasik) [reply rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictParamRector/Fixture/skip_parent_overridden.php.inc:13]: ok, I added parent already has return fixture https://github.com/rectorphp/rector-src/pull/4878/commits/397fc3aecd996b2bfbfc13399a50ce84605205e0

**Comment by samsonasik** [comment]
@TomasVotruba I am merging it ;)

**Pull request merged**